benchmarks/oss.py

```diff
@@ -64,6 +65,39 @@ class OptimType(str, Enum):
     everyone = "everyone"
 
 
+def validate_benchmark(measurements, args, check_regression):
+    """Validate the measurments against the golden benchmark config."""
+
+    golden_data = oss_mnist.get_golden_real_stats()
+
+    max_memory = -1.0
+    rank = dist.get_rank()
+    if not args.cpu:
+        # TODO(anj-s): Check if we need to synchronize before we caculate total training time.
+        torch.cuda.synchronize(rank)
+        max_memory = torch.cuda.max_memory_allocated(rank) / 2 ** 20
+        logging.info(f"[{rank}] : Peak memory {max_memory:.1f}MiB")
+
+    measurements.sort()
+    median = measurements[len(measurements) // 2]
+    # Compute the median and median of absolute differences img per second.
+    abs_diff = list(map(lambda x: abs(x - median), measurements))
+    abs_diff.sort()
+    mad = abs_diff[len(measurements) // 2] if args.epochs > 2 else -1
+
+    # TODO(anj-s): Is there value in logging this on every rank? Won't this be the same value (more or less)?
```

Comment:
at least for me it's useful to get the numbers locally whenever I do a change to shardedddp or oss, I use this quite often, it's as much a tool to get a first perf impression as it is a regression test.
per rank stats could be useful to have an idea of the dispersion in between the ranks, for instance if the sharding/balancing was very wrong then one rank could be significantly slower/faster